src/activation_token/gtk4.rs

```diff
@@ -0,0 +1,26 @@
+use gtk4::{glib::translate::*, prelude::*};
+
+use crate::ActivationToken;
+
+impl ActivationToken {
+    /// Gets an activation token from a window.
+    pub fn from_window(widget: &impl IsA<::gtk4::Widget>) -> Option<Self> {
+        let display = widget.as_ref().display();
+        let context = display.app_launch_context();
+
+        // FIXME Call the vfunc directly since
```

Comment:
Maybe we should wait for this MR to be merged. https://gitlab.gnome.org/GNOME/glib/-/merge_requests/3933